### DISCUSSION TITLE
After creating a bridge in Win11 observed a lot of TCP-retransmissions

### DISCUSSION THREAD

**Discussion by rhermsen** [microsoft/WSL#12034] [Q&A]
System details:
```
WSL2: Ubuntu 24.04.1
Win11: 23H2 build 22631.4037
WSL version: 2.2.4.0
Kernel: 5.15.153.1-microsoft-standard-WSL2
```

.wslconfig:
```
[wsl2]
networkingMode=bridged
vmSwitch=WSL
dhcp=true
```

I have a HTPC with 3 Ethernet ports, one connected to my Internet router, and two on one NIC currently unused.
To free up a port on the Internet router I would like to use one of the free ports to connect a device with access to the Internet router.
This HTPC also has a WSL2 instance active which via Hyper-V Manager > Virtual Switch manager connects the vNIC of the WSL2 instance to the Internet router.

Starting from a good working condition, I configured a bridge in Win11 and added the interface connected to Internet and an unused port on a NIC.
(commands below)
This looked to work ok.
- The WSL2 instance obtains its normal IP address via DHCP, and is still reachable via SSH
- Also the applications running in the WSL2 instance are reachable.
- The Win11 host is assigned a fix IP address and, with that, is also normally reachable.

But on a TCP connection between the WSL2 instance and the Win11 host get a lot of retransmissions during video-streaming (e.g. recording or live-tv).
This is also causing very frequent buffering in the video application (Kodi).

Is this a known limitation?
Or did I configure it incorrect?

Bridge configuration:
```
netsh bridge show adapter
netsh bridge create "vEthernet (WSL)" "Ethernet 2"
netsh interface ipv4 set address name="Network Bridge" static <static IP@> 255.255.255.0 <gw>
netsh interface ipv4 set dns name="Network Bridge" static <dns ip>
```

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
Sad news from just released [WSL v2.4.5](https://github.com/microsoft/WSL/releases/tag/2.4.5) changelog:

> Add user warning letting users know bridged networking will be deprecated

Bridged mode works very well as it is now in WSL-Version: 2.4.5.0
